jormungandr/src/fragment/pool.rs

```diff
@@ -93,6 +110,20 @@ impl Pool {
     }
 }
 
+fn is_fragment_valid(fragment: &Fragment) -> bool {
```

Comment:
Don't we have something already in chain-libs?

Reply:
I was looking for something like that, but couldn't find anything. All there was was witness count validation during deserialization, so I didn't have to check that and the balance check, which I used. Is there something better? Maybe we should and this check should end up in `chain-libs`.